### PR TITLE
Use official AWS SDK for S3

### LINKED ISSUE #14: Switch to a maintained S3 package
Since September 2014 `grunt-s3` [is unmaintained](https://github.com/pifantastic/grunt-s3/commit/110d0dfe0f61d3f6cf481dbfe3faad1c8613492d).

@pifantastic recommends switching to one of the following packages:

- https://github.com/MathieuLoutre/grunt-aws-s3
- https://github.com/jpillora/grunt-aws

### PR DESCRIPTION
**Pull request by thomas-kununu**
Closes #14

### CODE REVIEW

**Comment by thanpolas** [comment]
Thanks @thomas-kununu i'm gonna need some time to test this with current installation and i'll get back to you

File: lib/asset-s3-stat.js [hunk 1]
```diff
@@ -307,7 +333,7 @@ s3stat.Task.prototype._parseUploadFiles = function() {
 
     sourceFile = grunt.file.expand(assetObj.absPath)[0];
 
-    if (!sourceFile) {
+    if (!sourceFile || !assetObj.absPath.match(/(app\.min|ie\.min)/)) {
```
Comment (thanpolas) [inline lib/asset-s3-stat.js]: what is this supposed to do? It breaks upload on my local...

**Comment by thanpolas** [comment]
@thomas-kununu I've thoroughly tested the PR, it doesn't work as it is, the whole API has changed, we used to require the special `upload` attribute in the config which was then passed to the grunt-s3 task...

Now with that removed we have to devise a new API, I've seen you already used `options.headers` and treated it like an array... All of these breaking changes need to be documented in a changelog and documentation needs to be updated.

I've sent a PR to your PR with linting fixes and tweaks.

File: lib/asset-s3-stat.js (hunk 1 above)
Comment (thomas-kununu) [inline lib/asset-s3-stat.js]: Oh crap, that was not supposed to be here. That was just a temporary workaround for our project

**Comment by thanpolas** [comment]
thanks, so what's the plan with the `upload` attributes that need to be moved into `options`

**Comment by thomas-kununu** [comment]
I'm not sure I understand what you mean. Am I doing something wrong here? https://github.com/kununu/assetflow/blob/use-grunt-aws/lib/asset-s3-stat.js#L470-L482

**Comment by thanpolas** [comment]
No, that part is ok... we just have to update the docs and the changelog with the breaking change (`upload` gets moved into `options`). 

It's this part that worries me, you are accessing the `options.upload.header` as an array in `options.headers`... Any reason you moved that out of the `upload` params object?

We need to start documenting these changes

**Comment by thomas-kununu** [comment]
I've changed the header object to be referenced from the `assetObj` object.

I am now reviewing the various options from grunt-s3 that are not yet implemented in this branch. We'll need to add our own gzip compression before you can merge that branch. Thanks for your support for now.

**Comment by thomas-kununu** [comment]
I updated the readme file and implemented gzip compression now. Works fine for us, can you have a look at it @thanpolas ? Thx!

File: lib/asset-s3-stat.js [hunk 2]
```diff
@@ -247,31 +258,106 @@ s3stat.Task.prototype._startUpload = function( assetObj ) {
...
+      cb();
+      return def.resolve();
     }
-    def.resolve();
-  }.bind(this));
 
-  _def.fail(function(err) {
-    log.warn('Failed to upload: ' + assetObj.file.blue + ' to: ' +
-      assetObj.dest.blue + ' Error: ' + (err + '').red);
+    var header, value;
```
Comment (thanpolas) [inline lib/asset-s3-stat.js]: > `value` is defined but never used

Linting

File: lib/asset-s3-stat.js [hunk 3]
```diff
@@ -247,31 +258,106 @@ s3stat.Task.prototype._startUpload = function( assetObj ) {
...
+          } else {
+            if (this._progress) {
+              this._paceUpload.op();
+            }
+            def.resolve();
+          }
+        });
+      }
+    });
+  }
```
Comment (thanpolas) [inline lib/asset-s3-stat.js]: > Missing Semicolon

Linting

**Comment by thanpolas** [comment]
It's not yet clear which options remain from the migration from grunt-s3, more specifically, all the options that were previously available via grunt-s3 (http://than.pol.as/bApN), I see you've removed options `download` and `del` so that means that all the other options are supported?

If so, they all need to be re-written as all the other options parameters as seen right above: http://than.pol.as/bB0P

Once that part is cleared out, then the example from the docs needs to be updated (http://than.pol.as/bBdY) which now demonstrates how to configure using the `upload` keyword which belongs to grunt-s3.

Then these options need to be reflected [in the default options](https://github.com/verbling/assetflow/pull/16/files#diff-c2cebe855c5a8182160d2fdded917f67R499) of the `_getConfig()` method.

So to sum it up:

* [ ] Update README with which of the grunt-s3 options we now support and under which name, make them look like the other native options (http://than.pol.as/bB0P) 
* [ ] Update the configuration example, it uses legacy `upload` option
* [ ] Update the default options of `_getConfig()` method

I can do that for you but since you didn't merge my PR I understand you got it :)

Thank you for the great work!

**Comment by karlhorky** [comment]
Fixed the linting errors. We'll get to the other issues tomorrow.

**Comment by karlhorky** [comment]
If you have time, we'd be happy to merge a new PR with these changes.

Sorry about closing the old one - I thought there was an easier way if we added you as a collaborator to our fork, but that's not so straightforward with GitHub organizations unfortunately.
